presto-clp/src/main/java/com/facebook/presto/plugin/clp/optimization/ClpUdfRewriter.java

```diff
@@ -0,0 +1,303 @@
+        Map<ColumnHandle, VariableReferenceExpression> map = new HashMap<>();
+        root.getSources().forEach(source -> map.putAll(collectExistingScanAssignments(source)));
+        if (root instanceof TableScanNode) {
+            TableScanNode scan = (TableScanNode) root;
+            scan.getAssignments().forEach((var, handle) -> {
+                map.putIfAbsent(handle, var);
+            });
+        }
+        return map;
+    }
+
+    private class Rewriter
+            extends ConnectorPlanRewriter<Void>
+    {
+        private final PlanNodeIdAllocator idAllocator;
+        private final VariableAllocator variableAllocator;
+        private final Map<ColumnHandle, VariableReferenceExpression> globalColumnVarMap;
+
+        public Rewriter(
+                PlanNodeIdAllocator idAllocator,
+                VariableAllocator variableAllocator,
+                Map<ColumnHandle, VariableReferenceExpression> globalColumnVarMap)
+        {
+            this.idAllocator = idAllocator;
+            this.variableAllocator = variableAllocator;
+            this.globalColumnVarMap = globalColumnVarMap;
+        }
+
+        @Override
+        public PlanNode visitProject(ProjectNode node, RewriteContext<Void> context)
+        {
+            Assignments.Builder newAssignments = Assignments.builder();
+            for (Map.Entry<VariableReferenceExpression, RowExpression> entry : node.getAssignments().getMap().entrySet()) {
+                newAssignments.put(
+                        entry.getKey(),
+                        rewriteClpUdfs(entry.getValue(), functionManager, variableAllocator));
+            }
+
+            PlanNode newSource = rewritePlanSubtree(node.getSource());
+            return new ProjectNode(node.getSourceLocation(), idAllocator.getNextId(), newSource, newAssignments.build(), node.getLocality());
+        }
+
+        @Override
+        public PlanNode visitFilter(FilterNode node, RewriteContext<Void> context)
+        {
+            return buildNewFilterNode(node);
+        }
+
+        /**
+         * Rewrites <code>CLP_GET_*</code> UDFs in a {@link RowExpression}, collecting each resulting
+         * variable into the given map along with its associated {@link ColumnHandle}.
+         * <p>
+         * Each <code>CLP_GET_*</code> UDF must take a single constant string argument, which is used to
+         * construct the name of the variable reference (e.g. <code>CLP_GET_STRING('foo')</code> becomes
+         * a variable name <code>foo</code>). Invalid usages (e.g., non-constant arguments) will throw a
+         * {@link PrestoException}.
+         *
+         * @param expression the input expression to analyze and possibly rewrite
+         * @param functionManager function manager used to resolve function metadata
+         * @param variableAllocator variable allocator used to create new variable references
+         * @return a possibly rewritten {@link RowExpression} with <code>CLP_GET_*</code> calls replaced
```

Comment:
100 col wrappinga and minor update for the `@return`
```suggestion
         * Rewrites <code>CLP_GET_*</code> UDFs in a {@link RowExpression}, collecting each
         * resulting variable into the given map along with its associated {@link ColumnHandle}.
         * <p>
         * Each <code>CLP_GET_*</code> UDF must take a single constant string argument, which is
         * used to construct the name of the variable reference (e.g.
         * <code>CLP_GET_STRING('foo')</code> becomes a variable name <code>foo</code>). Invalid
         * usages (e.g., non-constant arguments) will throw a {@link PrestoException}.
         *
         * @param expression the input expression to analyze and possibly rewrite
         * @param functionManager function manager used to resolve function metadata
         * @param variableAllocator variable allocator used to create new variable references
         * @return a possibly rewritten {@link RowExpression} by <code>CLP_GET_*</code> calls
```